### PR TITLE
DEV-2860-build-artifact-for-silicon-mac

### PR DESCRIPTION
**Pull request by qqiao2024**

### CODE REVIEW

**Comment by qqiao2024** [comment]
macos-14 is much faster than previous version:
new version: 9.030 6.141 6.061
previous version: 17.643 16.775 17.035


```
❯ time ~/Downloads/gdc-client\ 3 -h
usage: gdc-client 3 [-h] [--version]
                    {download,upload,settings} ...

The Genomic Data Commons Command Line Client

optional arguments:
  -h, --help            show this help message and exit
  --version             show program's version number and exit

commands:
  {download,upload,settings}
                        for more information, specify -h after a
                        command
    download            download data from the GDC
    upload              upload data to the GDC
    settings            display default settings
~/Downloads/gdc-client\ 3 -h  0.47s user 0.28s system 8% cpu 9.030 total
❯ time ~/Downloads/gdc-client\ 3 -h
usage: gdc-client 3 [-h] [--version]
                    {download,upload,settings} ...

The Genomic Data Commons Command Line Client

optional arguments:
  -h, --help            show this help message and exit
  --version             show program's version number and exit

commands:
  {download,upload,settings}
                        for more information, specify -h after a
                        command
    download            download data from the GDC
    upload              upload data to the GDC
    settings            display default settings
~/Downloads/gdc-client\ 3 -h  0.46s user 0.27s system 11% cpu 6.141 total
❯ time ~/Downloads/gdc-client\ 3 -h
usage: gdc-client 3 [-h] [--version]
                    {download,upload,settings} ...

The Genomic Data Commons Command Line Client

optional arguments:
  -h, --help            show this help message and exit
  --version             show program's version number and exit

commands:
  {download,upload,settings}
                        for more information, specify -h after a
                        command
    download            download data from the GDC
    upload              upload data to the GDC
    settings            display default settings
~/Downloads/gdc-client\ 3 -h  0.46s user 0.27s system 12% cpu 6.061 total
❯ time ~/Downloads/gdc-client -h
usage: gdc-client [-h] [--version]
                  {download,upload,settings} ...

The Genomic Data Commons Command Line Client

optional arguments:
  -h, --help            show this help message and exit
  --version             show program's version number and exit

commands:
  {download,upload,settings}
                        for more information, specify -h after a
                        command
    download            download data from the GDC
    upload              upload data to the GDC
    settings            display default settings
~/Downloads/gdc-client -h  0.30s user 0.18s system 2% cpu 17.643 total
❯ time ~/Downloads/gdc-client -h
usage: gdc-client [-h] [--version]
                  {download,upload,settings} ...

The Genomic Data Commons Command Line Client

optional arguments:
  -h, --help            show this help message and exit
  --version             show program's version number and exit

commands:
  {download,upload,settings}
                        for more information, specify -h after a
                        command
    download            download data from the GDC
    upload              upload data to the GDC
    settings            display default settings
~/Downloads/gdc-client -h  0.28s user 0.15s system 2% cpu 16.775 total
❯ time ~/Downloads/gdc-client -h
usage: gdc-client [-h] [--version] {download,upload,settings} ...

The Genomic Data Commons Command Line Client

optional arguments:
  -h, --help            show this help message and exit
  --version             show program's version number and exit

commands:
  {download,upload,settings}
                        for more information, specify -h after a command
    download            download data from the GDC
    upload              upload data to the GDC
    settings            display default settings
~/Downloads/gdc-client -h  0.28s user 0.15s system 2% cpu 17.035 total
```

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ on:
       - develop
       - release/**
       - hotfix/**
+      - feat/dev-2860-build-artifact-for-silicon-mac
```
Comment (mpsolano) [inline .github/workflows/ci.yaml]: Will this be removed prior to merge?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -16,6 +17,7 @@ jobs:
         os:
           - macos-12
           - macos-13
+          - macos-14
```
Comment (mpsolano) [inline .github/workflows/ci.yaml:19]: OS14 is still compatible with some intel based macs. I'm assuming that it's optimized for apple silicon. In testing the binary that's produced, does this fix the lag that was being reported for m-series macs?

**Review by mpsolano — APPROVED** [review]
Looks good!

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (qqiao2024) [reply .github/workflows/ci.yaml:19]: It halved the time.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (qqiao2024) [reply .github/workflows/ci.yaml]: removed

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (mpsolano) [reply .github/workflows/ci.yaml:19]: Woohoo! Great work Qiao!

**Review by jiakf — APPROVED** [review]

**Pull request merged**